### ISSUE TITLE
`builtins.fetchTarball` leads to `Read-only filesystem` error

### ISSUE THREAD

**Issue by wizeman** [NixOS/nixops#291]
I'm trying to import (into one of my nixops-managed machines) the Hydra module from inside a source tarball, but all suggestions I've seen so far either lead to an infinite recursion error, or a `*.drv path is not valid` error (similar to the one in NixOS/nixpkgs#5996).

To solve this issue, I'm trying to use the new `builtins.fetchTarball` Nix built-in function (using Nix from a recent git commit).

My base configuration works fine in either `nixos-rebuild` or with `nixops`.
This is the new code I added to my machine's config:
```
    { config, pkgs, lib, ... }:

    let

>>>   hydra_src = builtins.fetchTarball "file:///home/tarballs/hydra.tar.gz";

    in {
      imports = [
        [...]
>>>     "${hydra_src}/hydra-module.nix"
      ];

    [...]
```

When I evaluate this new configuration using `nixos-rebuild`, everything goes smoothly.
However, when I try to evaluate the same configuration using `nixops`, this is what happens:

```
# nixops deploy --include homeserv --build-only --show-trace
downloading ‘file:///home/tarballs/hydra.tar.gz’...
error: while evaluating the attribute ‘config.deployment’ at /nixdev/nixpkgs/lib/attrsets.nix:135:44:
while evaluating anonymous function at /nixdev/nixpkgs/lib/options.nix:122:41, called from /nixdev/nixpkgs/lib/attrsets.nix:135:52:
while evaluating ‘scrubOptionValue’ at /nixdev/nixpkgs/lib/options.nix:118:22, called from /nixdev/nixpkgs/lib/options.nix:122:44:
while evaluating ‘isDerivation’ at /nixdev/nixpkgs/lib/attrsets.nix:222:18, called from /nixdev/nixpkgs/lib/options.nix:119:8:
while evaluating the attribute ‘config’ at /nixdev/nixpkgs/nixos/lib/eval-config.nix:50:5:
while evaluating the attribute ‘config’ at /nixdev/nixpkgs/lib/modules.nix:82:25:
while evaluating ‘yieldConfig’ at /nixdev/nixpkgs/lib/modules.nix:69:29, called from /nixdev/nixpkgs/lib/modules.nix:68:16:
while evaluating ‘mergeModules’ at /nixdev/nixpkgs/lib/modules.nix:160:26, called from /nixdev/nixpkgs/lib/modules.nix:59:17:
while evaluating ‘mergeModules'’ at /nixdev/nixpkgs/lib/modules.nix:164:36, called from /nixdev/nixpkgs/lib/modules.nix:161:5:
while evaluating ‘concatMap’ at /nixdev/nixpkgs/lib/lists.nix:54:18, called from /nixdev/nixpkgs/lib/modules.nix:200:9:
while evaluating ‘fold’ at /nixdev/nixpkgs/lib/lists.nix:20:19, called from /nixdev/nixpkgs/lib/modules.nix:59:38:
while evaluating ‘fold'’ at /nixdev/nixpkgs/lib/lists.nix:23:15, called from /nixdev/nixpkgs/lib/lists.nix:27:8:
while evaluating ‘closeModules’ at /nixdev/nixpkgs/lib/modules.nix:86:27, called from /nixdev/nixpkgs/lib/modules.nix:54:16:
while evaluating anonymous function at /nixdev/nixpkgs/lib/modules.nix:88:49, called from /nixdev/nixpkgs/lib/lists.nix:49:19:
opening lock file ‘/nix/store/4b9i4i8blv4jk8m8qkrsh722xpllmfq3-hydra.tar.gz.lock’: Read-only file system
error: evaluation of the deployment specification failed
```

Any ideas?

**Comment by wizeman** [comment]
Closing in favor of NixOS/nix#529.

**Comment by wizeman** [comment]
I'm reopening because it seems that `nix` provides a `--read-write-mode` option that fixes this issue.
Also, if I run `nixops` with the `--read-write-mode` option, then everything works fine.

So to be honest I'm no longer sure whether this is a bug.

**Comment by edolstra** [comment]
It's a current limitation of `fetchTarball`, but one that I want to address.

**Comment by domenkozar** [comment]
see #279

**Comment by aszlig** [comment]
Should be fixed by #279, please reopen if it still persists.
